tests/pass/libc.rs

```diff
@@ -42,7 +42,7 @@ fn test_posix_fadvise() {
     assert_eq!(result, 0);
 }
 
-#[cfg(any(target_os = "linux", target_os = "freebsd"))]
```

Comment:
Why disable these tests on freebsd?

Reply:
They don't even build on freebsd. They use Linux-only `libc`  symbols. I didn't catch this in https://github.com/rust-lang/miri/pull/2221.
FreeBSD currently only runs a subset of the test suite since the target is still WIP, and this file is not one of them. But I wanted to make sure at least it builds.